ci/test/00_setup_env_arm.sh

```diff
@@ -25,4 +25,4 @@ export RUN_FUNCTIONAL_TESTS=false
 export GOAL="install"
 # -Wno-psabi is to disable ABI warnings: "note: parameter passing for argument of type ... changed in GCC 7.1"
 # This could be removed once the ABI change warning does not show up by default
-export BITCOIN_CONFIG="--enable-reduce-exports --enable-suppress-external-warnings --enable-werror CXXFLAGS=-Wno-psabi --with-boost-process"
+export BITCOIN_CONFIG="--enable-reduce-exports --enable-suppress-external-warnings CXXFLAGS=-Wno-psabi --with-boost-process"
```

Comment:
`export NO_WERROR=1` is not supposed to be set for tsan/ubsan jobs.

it fails, because both tsan and ubsan jobs use `clang` but all other jobs use gcc. Somehow, after `configure` there are set incorrect options:
```
clang++-16 -std=c++17 -c -pipe -static-libstdc++ -O1   -Werror -D_GLIBCXX_DEBUG -D_GLIBCXX_DEBUG_PEDANTIC -D_LIBCPP_DEBUG=1 -I/builds/dashpay/dash/depends/x86_64-pc-linux-gnu/include/  conftest.cpp
```
clang doesn't support `-static-llibstdc++` which is supposed to be gcc-only, it cause this failure:
```
clang: error: argument unused during compilation: '-static-libstdc++' [-Werror,-Wunused-command-line-argument]
```
This failure make `configure.ac` think that `clang` doesn't suppor -Werror and fails with error. So, you can't activate this flag. 

I excluded this backport from this PR